lib/stdlib/src/supervisor.erl

```diff
@@ -1521,16 +1521,25 @@ shutdown(#child{pid=Pid, shutdown=Time} = Child) ->
         end
     end.
 
-unlink_flush(Pid, DefaultReason) ->
-    %% We call unlink in order to guarantee that the 'EXIT' has arrived
-    %% from the dead process. See the unlink docs for details.
+unlink_flush(Pid, noproc) ->
+    {links, Ls} = process_info(self(),links),
+    Timeout = case lists:member(Pid, Ls) of
+                  true -> infinity;
+                  false -> 0
+              end,
+    receive
+        {'EXIT', Pid, ExitReason} ->
+            ExitReason
+    after Timeout ->
+            naughty_child
```

Comment:
Didn't we decide on some other reason in this case? `child_process_unlinked` or something similar.

Reply:
Yes I think we did. @u3s we should fix that.

Reply:
fixed